cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/templates/etl/common/StageConfigurator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.common;
+
+import co.cask.cdap.api.templates.AdapterConfigurer;
+import co.cask.cdap.templates.etl.api.EndPointStage;
+import co.cask.cdap.templates.etl.api.PipelineConfigurer;
+import co.cask.cdap.templates.etl.api.StageSpecification;
+import co.cask.cdap.templates.etl.api.Transform;
+import co.cask.cdap.templates.etl.api.config.ETLStage;
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+
+import java.util.List;
+
+/**
+ * Configurator utility class that has helper methods to configure Source/Sink and Transform stages.
+ */
+public class StageConfigurator {
+  private static final Gson GSON = new Gson();
+
+  public static void configure(EndPointStage stage, ETLStage stageConfig, AdapterConfigurer configurer,
```

Comment:
what do you think of moving this type of logic into a ```configurePipeline(AdapterConfigurer configurer)``` method on a base ETLConfig class that both ETLBatchConfig and ETLRealtimeConfig extend? That class could have source, transforms, and sink configs since they are common for both batch and realtime.  Then the configuration and instantiation logic is all in one place instead of spread out across this utility class and the drivers.